packages/gui/src/components/nfts/NFTContextualActions.tsx

```diff
@@ -558,6 +564,77 @@ function NFTBurnContextualAction(props: NFTBurnContextualActionProps) {
     </MenuItem>
   );
 }
+
+/* ========================================================================== */
+/*                     Invalidate cache of a single NFT                       */
+/* ========================================================================== */
+
+type NFTInvalidateContextualActionProps = NFTContextualActionProps;
+
+function NFTInvalidateContextualAction(
+  props: NFTInvalidateContextualActionProps,
+) {
+  const { onClose, selection } = props;
+
+  const selectedNft: NFTInfo | undefined = selection?.items[0];
+  const disabled = !selectedNft || selectedNft?.pendingTransaction;
+  const dataUrl = selectedNft?.dataUris?.[0];
+  const [, setThumbCache] = useLocalStorage(
+    `thumb-cache-${selectedNft.$nftId}`,
+    null,
+  );
+  const [, setContentCache] = useLocalStorage(
+    `content-cache-${selectedNft.$nftId}`,
+    null,
+  );
+
+  const [forceReloadNFT, setForceReloadNFT] = useLocalStorage(
+    `force-reload-${selectedNft.$nftId}`,
+    false,
+  );
+
+  const [, setMetadataCache] = useLocalStorage(
+    `metadata-cache-${selectedNft.$nftId}`,
+    {},
+  );
+
+  async function handleInvalidate() {
+    if (!selectedNft) {
+      return;
+    }
+    setThumbCache({});
+    setContentCache({});
+    setMetadataCache({});
+    setForceReloadNFT(!forceReloadNFT);
+    const ipcRenderer = (window as any).ipcRenderer;
+    ipcRenderer.invoke(
+      'removeCachedFile',
+      computeHash(`${selectedNft.$nftId}_${dataUrl}`, { encoding: 'utf-8' }),
+    );
+  }
+
+  if (!dataUrl) {
+    return null;
+  }
+
+  return (
+    <MenuItem
+      onClick={() => {
+        onClose();
+        handleInvalidate();
+      }}
+      disabled={disabled}
+    >
```

Comment:
Since onClose doesn't exist:

```suggestion
    <MenuItem onClick={handleInvalidate} disabled={disabled} close>
```